### PR TITLE
feat: render query explanation in tree like format

### PR DESCRIPTION
**Pull request by rin-yato**
Not exactly sure if this is the way to do it, but here it is 😆

https://github.com/user-attachments/assets/e4e29949-8ad1-4037-ad89-ebd4894a9431

### CODE REVIEW

**Comment by invisal** [comment]
Look nice. Will check :)

**Comment by rin-yato** [comment]
> Look nice. Will check :)

I'll do some clean up tmr, just checked the ci log

File: src/components/gui/query-explain.tsx [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+export interface ExplainRow {
+  id: number;
+  parent: number;
+  notused: number;
+  detail: string;
+}
+
+export type ExplainRowWithChildren = ExplainRow & {
+  children?: ExplainRowWithChildren[];
+};
```
Comment (invisal) [inline src/components/gui/query-explain.tsx:18]: Everything looks solid. But since we don't use it somewhere else, there is no reason to export it out

**Review by invisal — CHANGES_REQUESTED** [review]

File: src/components/gui/query-explain.tsx (hunk 1 above)
Comment (rin-yato) [reply src/components/gui/query-explain.tsx]: Pushed an update. I cleaned up some stuffs, remove unnecessary exports and use consistent naming

**Review by invisal — APPROVED** [review]
LGTM

**Review by invisal — APPROVED** [review]
LGTM

**Pull request merged**